File: src/main/java/com/ryanpmartz/kupload/web/form/RegistrationForm.java
package com.ryanpmartz.kupload.web.form;

import org.hibernate.validator.constraints.Email;
import org.hibernate.validator.constraints.NotBlank;

import javax.validation.constraints.Size;

public class RegistrationForm {

    @NotBlank(message = "Please enter your first name")
    private String firstName;

    @NotBlank(message = "Please enter your last name")
    private String lastName;

    @Email(message = "Please enter a valid email address")
    private String email;

    @NotBlank(message = "Please enter a password")
    @Size(min = 8, message = "Password must be at least 8 characters")
    private String password;

    @NotBlank(message = "Please re-enter password for confirmation")
    @Size(min = 8, message = "Password must be at least 8 characters")
    private String confirmPassword;

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public String getConfirmPassword() {
        return confirmPassword;
    }

    public void setConfirmPassword(String confirmPassword) {
        this.confirmPassword = confirmPassword;
    }
}

File: src/main/java/com/ryanpmartz/kupload/web/form/RegistrationForm.kt
package com.ryanpmartz.kupload.web.form

import org.hibernate.validator.constraints.Email
import org.hibernate.validator.constraints.NotBlank

import javax.validation.constraints.Size

class RegistrationForm {

    @NotBlank(message = "Please enter your first name")
    var firstName: String? = null

    @NotBlank(message = "Please enter your last name")
    var lastName: String? = null

    @Email(message = "Please enter a valid email address")
    var email: String? = null

    @NotBlank(message = "Please enter a password")
    @Size(min = 8, message = "Password must be at least 8 characters")
    var password: String? = null

    @NotBlank(message = "Please re-enter password for confirmation")
    @Size(min = 8, message = "Password must be at least 8 characters")
    var confirmPassword: String? = null
}
